api/controllers/topics.controllers.js

```diff
@@ -1,7 +1,18 @@
-const { selectTopics } = require("../models/topics.models");
+const { selectTopics, selectArticleById } = require("../models/topics.models");
 
 exports.getTopics = (req, res, next) => {
 	selectTopics().then(({ rows }) => {
 		res.status(200).send({ topics: rows });
 	});
 };
+
+exports.getArticleById = (req, res, next) => {
+	selectArticleById(req)
+		.then(({ rows }) => {
+			if (!rows.length) {
+				return Promise.reject({ status: 404 });
```

Comment:
The rows should be destructured in the model ideally. We just want the controller to essentially invoke the model, then send the response. The model should therefore deal with the promise.reject as well. 👍